Dockerfile

```diff
@@ -19,10 +19,14 @@ FROM base as build
 
 # Install packages needed to build gems and node modules
 RUN apt-get update -qq && \
-    apt-get install --no-install-recommends -y build-essential curl git libpq-dev libvips node-gyp pkg-config python-is-python3
+    apt-get install --no-install-recommends -y \
+    build-essential \
+    curl \
+    libmariadb-dev  `# Needed by ActiveRecord` \
```

Comment:
I know `libvips` is used by ActiveRecord. `node-gyp` and `pkg-config` are used for building native Node modules (possibly the python one too), which are a part of the asset pipeline IIRC.

I'm confused how Rails recommends the SQL client lib for MariaDB, though for our version we definitely needs `libmariadb-dev`. See the last link below, but they don't install a native package for MariaDB (or maybe install the MySQL client which works too?). I suppose we could run `rails new` and tell it use to MariaDB and see what it spits out.

Ideally we stick to the Rails-suggested Dockerfile templates as much as possible, This is to make Rails version upgrades much, much easier and so won't have unexpected issues (i.e. implicit dependency provided by another package changes). When we run `rails app:update` after hitting Rails 7 it will suggest changes to this file.

See these files:
https://github.com/rails/rails/blob/3769f1eee6a1ec4f6438155d67477db6d192577f/railties/lib/rails/generators/app_base.rb#L589
https://github.com/rails/rails/blob/3769f1eee6a1ec4f6438155d67477db6d192577f/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L18
https://github.com/rails/rails/blob/3769f1eee6a1ec4f6438155d67477db6d192577f/railties/lib/rails/generators/database.rb#L173

Reply:
I didn't have any issues during build or basic testing after removing these, but I haven't run further tests. I'll restore it to Rails defaults for compatibility.

I'm not sure if Rails needs it for connecting to MariaDB as `mariadb://`. However our development steps use MySQL, and we use the mysql2 gem, which relies on mariadb dev tools.